src/app/page/Missions/Missions.jsx

```diff
@@ -40,41 +40,41 @@ const MissionsPage = ({ user, history, firebase, ...rest }) => {
     }
   }
 
-  if (!missions) {
-    return <div> isloading...</div>;
-  }
   return (
     <Page template="pink">
       <StyledHeader variant="h1"> Missions </StyledHeader>
+      {missions ? (
+        missions.map((mission) => (
+          <Card key={mission.id}>
+            <MissionCard mission={mission} key={`preview-${mission.id}`} />
 
-      {missions.map((mission) => (
-        <Card key={mission.id}>
-          <MissionCard mission={mission} key={`preview-${mission.id}`} />
-
-          <Grid container justify="center" alignItems="center">
-            <StyledButton
-              color="primary"
-              size="large"
-              variant="contained"
-              disableElevation
-              onClick={() => volunteerForMission(mission.id)}
-            >
-              Volunteer
-            </StyledButton>
-            <PlaceHolder />
-            <StyledButton
-              variant="outlined"
-              size="large"
-              color="secondary"
-              onClick={() => {
-                history.push(`/missions/${mission.id}`);
-              }}
-            >
-              Details
-            </StyledButton>
-          </Grid>
-        </Card>
-      ))}
+            <Grid container justify="center" alignItems="center">
+              <StyledButton
+                color="primary"
+                size="large"
+                variant="contained"
+                disableElevation
+                onClick={() => volunteerForMission(mission.id)}
+              >
+                Volunteer
+              </StyledButton>
+              <PlaceHolder />
+              <StyledButton
+                variant="outlined"
+                size="large"
+                color="secondary"
+                onClick={() => {
+                  history.push(`/missions/${mission.id}`);
+                }}
+              >
+                Details
+              </StyledButton>
+            </Grid>
+          </Card>
+        ))
+      ) : (
+        <CircularProgress />
```

Comment:
This is good, but in a situation where there are errors fetching missions, the loading will be there forever until refreshed. Do we wanna keep a loading and error state to mitigate this behaviour??

Reply:
I think we have to.  Maybe create a separate issue for that?

Reply:
Correct, Felt that any error handling would creep the scope of this ticket. Was planning to bring that up in another issue.

The missions object looks to be assigned through a useSelector hook(?) that I'm I hadn't had a chance to get familiar with yet.

Would you advise to change the scope of the ticket to include error handling for the api call?

Reply:
Can you create an issue for that whilst I approve this